### DISCUSSION TITLE
How do I make the Hindsight starts automatically?

### DISCUSSION THREAD

**Discussion by gmooklee** [gopro/labs#261] [Q&A] [answered]
How do I make Hindsight automatically start when I turn on the GoPro 10? I activate the function manually after I turn on the camera. It is frustrating. I have a body camera from Transcend with a buffer recording feature that is the same as this Hindsight, but the difference is that I don't have to activate the feature manually. It starts the function when I turn on the camera.

**Comment by dnewman-gpsw** [comment, marked as answer]
You can do that with a boot command, any manual feature can be made automatic this way.  

They are two ways to use a boot command:
1) The brute force way: !MBOOT="hS1"  - this works fine if you know what you are doing.  However, if you make a typo, you can crash/lock your camera on boot (not an easy fix)
2) Adding a SD card based boot file: !MBOOT="!Lboot"!SAVEboot=hS1  - this is safest, as if the command is in error, eject the SD and it will not run.

PS. hS1 = hindsight 15s.

**Reply by gmooklee** [reply]
Thank you, I will try the second one. Can I ask one more question? What if I want 30s hindsight? Maybe it is hS2?

**Reply by dnewman-gpsw** [reply]
Yes, hS2 is 30s

**Reply by gmooklee** [reply]
Thank you, sir!

**Reply by gmooklee** [reply]
I tried the second one, but I still have to touch the Hindsight icon to activate the feature after I turn on my GoPro. Maybe that is impossible. The options is there, but this was there already.

**Reply by dnewman-gpsw** [reply]
It is working, however there is a bug in writing hS1 or hS2 with the !SAVE Command.  However, you manual edit your SD card's MISC/boot text file with hS2, it works fine.

**Reply by gmooklee** [reply]
God bless you, sir. It is working.

**Reply by DinnerBug** [reply]
Hello,
I tried creating a "boot.txt" file in my MISC folder on the Gopro's sd card but the script won't work.

**Reply by dnewman-gpsw** [reply]
If you want to use "boot.txt" you need this command !MBOOT="!Lboot.txt"

**Reply by DinnerBug** [reply]
Thank you very much!
